### PR TITLE
Java: update the url-redirection in the same style as the C# qhelp

### PR DESCRIPTION
**Pull request by erik-krogh**
I recently did a similar QHelp update for C#: https://github.com/github/codeql/pull/15623 

I found that the existing QHelp is not a good help for developers fixing url-redirection vulnerabilities.  
This PR mostly adds more fix-examples to the QHelp, because the existing example was rarely relevant to real vulnerabilities.  

The new fix examples should produce TNs now that https://github.com/github/codeql/pull/15731 has been merged.

### CODE REVIEW

**Review by atorralba — COMMENTED** [review]
LGTM, let's see what docs has to say.

File: java/ql/src/Security/CWE/CWE-601/UrlRedirect.qhelp [hunk 1]
```diff
@@ -16,21 +16,53 @@ controlled by the attacker.</p>
...
 
 </example>
-<references>
 
+<references>
 
+<li>
+OWASP:
+<a href="https://cheatsheetseries.owasp.org/cheatsheets/Unvalidated_Redirects_and_Forwards_Cheat_Sheet.html">XSS
+Unvalidated Redirects and Forwards Cheat Sheet</a>.
```
Comment (jcogs33) [inline java/ql/src/Security/CWE/CWE-601/UrlRedirect.qhelp:61]: Is `XSS` a typo here?
```suggestion
<a href="https://cheatsheetseries.owasp.org/cheatsheets/Unvalidated_Redirects_and_Forwards_Cheat_Sheet.html">Unvalidated Redirects and Forwards Cheat Sheet</a>.
```

Comment (erik-krogh) [reply java/ql/src/Security/CWE/CWE-601/UrlRedirect.qhelp:61]: Yes. And the same thing was in the C# QHelp, so I also fixed it there.

**Review by michaelnebel — APPROVED** [review]
C# LGTM!

File: java/ql/src/Security/CWE/CWE-601/UrlRedirect.qhelp [hunk 2]
```diff
@@ -16,21 +16,53 @@ controlled by the attacker.</p>
 <p>To guard against untrusted URL redirection, it is advisable to avoid putting user input
 directly into a redirect URL. Instead, maintain a list of authorized
 redirects on the server; then choose from that list based on the user input provided.</p>
-
+<p>
+If this is not possible, then the user input should be validated in some other way,
+for example, by verifying that the target URL is on the same host as the current page.
```
Comment (subatoi) [inline java/ql/src/Security/CWE/CWE-601/UrlRedirect.qhelp:21]:
```suggestion
If this is not possible, then the user input should be validated in some other way. For example, by verifying that the target URL is on the same host as the current page.
```

**Review by subatoi — APPROVED** [review]
👍 from Docs: only one tiny suggestion!

File: java/ql/src/Security/CWE/CWE-601/UrlRedirect.qhelp (hunk 2 above)
Comment (erik-krogh) [reply java/ql/src/Security/CWE/CWE-601/UrlRedirect.qhelp:21]: I'll skip. The existing phrasing is already used elsewhere.

**Pull request merged**